### DISCUSSION TITLE
k3s.service keeps getting "reset"

### DISCUSSION THREAD

**Discussion by analogrelay** [k3s-io/k3s#4333] [Q&A] [answered]
I've found I have to add `--flannel-backend=host-gw` to `k3s.service` to get networking working properly in my 4-node Raspberry Pi cluster. That's fine, it's working well. However, every time I restart the primary node, the setting is removed from `k3s.service`. Am I adding it in the wrong place? Is there a different way to properly persist that setting?

Every time I reboot the cluster I have to SSH in to the primary, add the setting, restart k3s, then SSH to each agent and restart k3s-agent. It's a bit of a pain and definitely feels like the "wrong" way 🤣.

I'm running on Raspberry Pi using Ubuntu 21.04

**Comment by brandond** [comment, marked as answer]
Where did you put the setting? Are you by some odd chance re-running the k3s install script every time your node starts up?

**Reply by analogrelay** [reply]
> Where did you put the setting?

I hand-edited the `/etc/systemd/system/k3s.service` file, and that very much might be the Wrong Way ™️ , since I'm a bit of a systemd and k3s noob ;P.

> Are you by some odd chance re-running the k3s install script every time your node starts up?

I don't _think_ I am. I just installed k3s by running the [install script](https://rancher.com/docs/k3s/latest/en/quick-start/) once.

It's entirely likely this is something on my system (hence why I started with a Discussion rather than an Issue 😆). Sounds like there isn't anything obvious in k3s itself that would rewrite this file.

**Reply by brandond** [reply]
no, that's the right place to do it. There's not anything in K3s itself that would modify that file after installation though, so it must be something else.

**Reply by analogrelay** [reply]
Thanks for that clarification! I'll dig deeper on my system then.
